app/screens/likerland-oauth-screen/likerland-oauth-screen.tsx

```diff
@@ -19,34 +24,44 @@ import { RootStore } from "../../models/root-store"
 
 import { translate } from "../../i18n"
 import { COMMON_API_CONFIG } from "../../services/api/api-config"
+import { logError } from "../../utils/error"
 
 export interface LikerLandOAuthScreenProps extends NavigationScreenProps<{}> {
   rootStore: RootStore,
 }
 
 @inject("rootStore")
-export class LikerLandOAuthScreen extends React.Component<LikerLandOAuthScreenProps, {}> {
-  private onNavigationStateChange = async (navState: WebViewNavigation) => {
-    const { url } = navState
-    const { rootStore } = this.props
+export class LikerLandOAuthScreen extends React.Component<LikerLandOAuthScreenProps> {
+  private handleError = async () => {
+    await this.props.rootStore.userStore.logout()
+    this.props.rootStore.userStore.setIsSigningIn(false)
+    this.props.navigation.goBack()
+  }
+
+  private handlePostSignIn = async () => {
+    await Promise.all([
+      this.props.rootStore.userStore.fetchUserInfo(),
+      this.props.rootStore.userStore.fetchLikerLandUserInfo(),
+    ])
+    this.props.navigation.navigate("App")
+
+    // Try to open the deferred deep link URL after sign in
+    this.props.rootStore.openDeepLink()
+  }
+
+  private handleURLChange = (url: string) => {
     if (url.includes("/following")) {
-      await Promise.all([
-        this.props.rootStore.userStore.fetchUserInfo(),
-        this.props.rootStore.userStore.fetchLikerLandUserInfo(),
-      ])
-      this.props.navigation.navigate("App")
-
-      // Try to open the deferred deep link URL after sign in
-      rootStore.openDeepLink()
+      this.handlePostSignIn()
+    } else if (Platform.OS === "android" && url.includes("/oauth/redirect")) {
```

Comment:
can we assume ios would finish the redirect on time too and set the same timer?
Would need to clear the timer if the event above fired first though